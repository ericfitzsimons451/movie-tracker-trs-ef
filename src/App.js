import React, { Component } from 'react';
import './App.css';
import './allPurposeFetch'
import  allPurposeFetch  from './allPurposeFetch';
import apiKey from './APIkey.js'
import { Route } from 'react-router-dom'
import AllMovies from './AllMovies' 

class App extends Component {
  constructor() {
    super() 
    this.state = {
      movies: [],
      //DOES THE STORE NEED TO KNOW ABOUT MOVIES?  OR, JUST USERS?  WHY?
      favorites: [],
      //there should be a TOGGLE function.  
      users: []
      //Almost gauranteed that we'll need the Redux store for these
    }
  }

  componentDidMount() {
    this.fetchMovies()
  }

  fetchMovies = async () => {
    const url = `https://api.themoviedb.org/3/movie/popular?api_key=${apiKey}&language=en-US&page=1`
    const movies = await allPurposeFetch(url)
    this.setState({ movies: movies.results })
    
    // this.displayMovies(this.state.movies)
  }

  displayMovies = (movieArray) => {
    this.state.movies.map((movie) => {

    })
  }

  render() {
    return (
      <div className="App">
        {/* <Header />
        <Nav /> */}
        <Route exact path='/' render={ () => <AllMovies movies={this.state.movies} />} />
      </div>
    );
  }
}

export default App;
