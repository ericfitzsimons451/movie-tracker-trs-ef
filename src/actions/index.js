const storeMovies = (movies) => ({
    type: 'STORE_MOVIES',
    movies,
})

export default storeMovies;