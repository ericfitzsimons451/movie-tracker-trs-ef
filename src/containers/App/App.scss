.App {
  margin: auto;
  height: 100%;
  width: 100%;
  display: grid;
  grid-template-areas:
    "header-container" 
    "display";
  background-color: black;
}

.header-container, 
.movie-display-container,
.display {
  max-width: 1500px;
  margin: 0px 25px;
}

.header-container {
  display: grid;
  grid-template-rows: 125px;
  grid-template-columns: 1fr 450px;
}

.display {
  grid-area: display;
}

