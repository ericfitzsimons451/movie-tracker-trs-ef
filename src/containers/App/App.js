import React, { Component } from 'react';
import './App.scss'
import  allPurposeFetch  from '../../allPurposeFetch';
import apiKey from '../../APIkey.js'
import { Route } from 'react-router-dom'
import AllMovies from '../../components/AllMovies/AllMovies' 
import Header from '../../components/Header/Header'
import Nav from '../../components/Nav/Nav'

class App extends Component {
  constructor() {
    super() 
    this.state = {
      movies: [],
      //DOES THE STORE NEED TO KNOW ABOUT MOVIES?  OR, JUST USERS?  WHY?
      favorites: [],
      //there should be a TOGGLE function.  
      users: []
      //Almost gauranteed that we'll need the Redux store for these
    }
  }

  componentDidMount() {
    this.fetchMovies()
  }

  fetchMovies = async () => {
    const url = `https://api.themoviedb.org/3/movie/popular?api_key=${apiKey}&language=en-US&page=1`
    const movies = await allPurposeFetch(url)
    this.setState({ movies: movies.results })
    
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Nav />
        <Route exact path='/' render={ () => <AllMovies movies={this.state.movies} />} />
      </div>
    );
  }
}

export default App;
