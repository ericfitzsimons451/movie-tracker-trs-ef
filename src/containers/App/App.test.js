import { App, mapDispatchToProps, mapStateToProps} from "./App";
import { storeMovies, storeUsers } from "../../actions";
import mockFetchedData from "../../mockData";
import { shallow } from "enzyme";
import React from "react";

describe("App", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);

  })
  describe("App", () => {
    it('should render the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  });

  describe("cleanMovies", () => {
    it("should take in an array of movies and return a cleaned array", () => {
      const movies = mockFetchedData;
      const expected = [
        {
          name: "How to Train Your Dragon: The Hidden World",
          id: 166428,
          poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
          release_date: "2019-01-03",
          vote_average: 7.8,
          overview:
            "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind."
        },
        {
          name: "Master Z: Ip Man Legacy",
          id: 450001,
          poster_path: "/2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
          release_date: "2018-12-20",
          vote_average: 5.5,
          overview:
            "After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect."
        }
      ];

      const results = wrapper.instance().cleanMovies(movies);
      expect(results).toEqual(expected);
    });
  });

  describe('cleanUsers', () => {
    it('should accept an array of users and return an array of emails', () => {
      const mockUsers = [
        {
          id: 1, 
          name: "Taylor", 
          password: "password", 
          email: "tman2272@aol.com"
        }, {
          id: 2, 
          name: "James", 
          password: "123", 
          email: "james@gmail.com"
        }
      ]
        const expected = ['tman2272@aol.com', 'james@gmail.com']
        const results = wrapper.instance().cleanUsers(mockUsers)
        expect(results).toEqual(expected)
    })
  })

  describe('mapStateToProps', () => {
    const mockState = {
      movies:[
        {
          name: "How to Train Your Dragon: The Hidden World",
          id: 166428,
          poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
          release_date: "2019-01-03",
          vote_average: 7.8,
          overview:
            "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind."
        },
        {
          name: "Master Z: Ip Man Legacy",
          id: 450001,
          poster_path: "/2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
          release_date: "2018-12-20",
          vote_average: 5.5,
          overview:
            "After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect."
        }
      ],
      favorites: []
    }
    const expected = {
      movies:[
        {
          name: "How to Train Your Dragon: The Hidden World",
          id: 166428,
          poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
          release_date: "2019-01-03",
          vote_average: 7.8,
          overview:
            "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind."
        },
        {
          name: "Master Z: Ip Man Legacy",
          id: 450001,
          poster_path: "/2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
          release_date: "2018-12-20",
          vote_average: 5.5,
          overview:
            "After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect."
        }
      ],
      favorites: []
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })

  describe("mapDispatchToProps", () => {
    it("should call dispatch with action.storeMovies and a payload of movies", () => {
      //setup
      const mockDispatch = jest.fn()
      const mockMovies = [{ title: "Aquaman" }, { title: "Triple Frontier" }];
      const actionToDispatch = storeMovies(mockMovies);
      const mappedProps = mapDispatchToProps(mockDispatch)

      //execution
      mappedProps.storeMovies(mockMovies);

      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch with action.storeUsers and a payload of users', () => {
  //     const mockDispatch = jest.fn()  
  //     const mockUsers = [
  //         {
  //           id: 1, 
  //           name: "Taylor", 
  //           password: "password", 
  //           email: "tman2272@aol.com"
  //         }, {
  //           id: 2, 
  //           name: "James", 
  //           password: "123", 
  //           email: "james@gmail.com"
  //         }
  //       ]
  //       const actionToDispatch = storeUsers(mockUsers)
  //       const mappedProps = mapDispatchToProps(mockDispatch)
  //       mappedProps.storeUsers(mockUsers)
  //       expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  //     })
  //   })
  // })

  describe("fetchMovies", () => {});
});
