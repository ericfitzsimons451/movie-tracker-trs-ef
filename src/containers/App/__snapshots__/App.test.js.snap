// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should render the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="header-container"
  >
    <Header />
    <withRouter(Connect(Nav)) />
  </div>
  <Route
    className="display"
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    className="display"
    path="/login"
    render={[Function]}
  />
  <Route
    className="display"
    exact={true}
    path="/favorites"
    render={[Function]}
  />
</div>
`;
